sentinelhub/time_utils.py

```diff
@@ -147,34 +161,80 @@ def parse_time_interval(time):
     :param time: An input time
     :type time: str or datetime.datetime or (str, str) or (datetime.datetime, datetime.datetime)
     :return: interval of start and end date of the form `YYYY-MM-DDThh:mm:ss`
-    :rtype: (str, str)
+    :rtype: (datetime.datetime, datetime.datetime)
     :raises: ValueError
     """
     if time == SHConstants.LATEST:
```

Comment:
Why 1985?

Reply:
Ah, this in order to get an entire interval of possibly available data. It is quite convoluted but it enables using `time='latest'` for OGC and Process API requests. Although now that I think about it OGC request will provide only the latest acquisition while Process API will provide a mosaic of acquisitions starting with the latest.

Perhaps it would be better to limit "latest" functionality only to OGC requests and remove this case from `parse_time_interval`?

Reply:
Limiting to OGC is a must, perhaps it would be prudent to remove it completely. Better that the users are explicit to what data they are looking for...

Reply:
Ok, I'll look into this.

Reply:
After sleeping it over it is now clear to me that this is a good decision. So I removed support for `'latest'` from `parse_time_interval` and only added handling of this in `WebFeatureService` class. Now only WMS, WCS, and WFS functionalities support this option.

I wouldn't completely remove it since this has been a default option of `WmsRequest` and `WcsRequest` since the beginning and things would break if we changed that.

Reply:
I agree with the changes.